pkg/tests/ossm/prometheus_scoped_scraping_test.go

```diff
@@ -0,0 +1,260 @@
+					"Found no permission error",
+					"Expected to find no permission error, but got some error",
+				),
+			)
+		}
+
+		checkConfigurationReloadingTriggered := func(t test.TestHelper, start time.Time) {
+			retry.UntilSuccess(t, func(t test.TestHelper) {
+				oc.LogsSince(t,
+					start,
+					prometheusPodSelector, "config-reloader",
+					assert.OutputContains("Reload triggered",
+						"Triggered configuration reloading",
+						"Expected to trigger configuration reloading, but did not",
+					),
+				)
+			})
+		}
+
+		testPrometheusConfigWithAsserts := func(t test.TestHelper, asserts ...common.CheckFunc) {
+			retry.UntilSuccess(t, func(t test.TestHelper) {
+				shell.Execute(t,
+					fmt.Sprintf("oc -n %s get configmap prometheus -o jsonpath='{.data.prometheus\\.yml}'", meshNamespace),
+					asserts...)
+			})
+		}
+
+		DeployControlPlane(t)
+
+		checkPermissionError(t)
+
+		t.NewSubTest("when creating a SMMR").Run(func(t test.TestHelper) {
+			ns := generateNamespace()
+
+			t.Cleanup(func() {
+				oc.DeleteNamespace(t, ns)
+				restoreDefaultSMMR(t)
+			})
+
+			t.LogStepf("Create namespace %s and add it into SMMR", ns)
+			oc.CreateNamespace(t, ns)
+			updateDefaultSMMRWithNamespace(t, ns)
+
+			t.LogStepf("Look for %s in prometheus ConfigMap", ns)
+			testPrometheusConfigWithAsserts(t, assertConfigMapContainsNamespace(ns))
+		})
+
+		t.NewSubTest("when adding a new namespace into existing SMMR").Run(func(t test.TestHelper) {
+			ns := generateNamespace()
+			anotherNs := generateNamespace()
+
+			t.Cleanup(func() {
+				oc.DeleteNamespace(t, ns, anotherNs)
+				restoreDefaultSMMR(t)
+			})
+
+			t.LogStepf("Create namespace %s and add it into SMMR", ns)
+			oc.CreateNamespace(t, ns)
+			updateDefaultSMMRWithNamespace(t, ns)
+
+			t.LogStepf("Create namespace %s and add it into SMMR along with %s", anotherNs, ns)
+			oc.CreateNamespace(t, anotherNs)
+			updateDefaultSMMRWithNamespace(t, ns, anotherNs)
+
+			t.LogStepf("Look for %s in prometheus ConfigMap", []string{ns, anotherNs})
+			testPrometheusConfigWithAsserts(t, assertConfigMapContainsNamespace(ns), assertConfigMapContainsNamespace(anotherNs))
+		})
+
+		t.NewSubTest("when removing a namespace from existing SMMR").Run(func(t test.TestHelper) {
+			ns := generateNamespace()
+			anotherNs := generateNamespace()
+
+			t.Cleanup(func() {
+				oc.DeleteNamespace(t, ns, anotherNs)
+				restoreDefaultSMMR(t)
+			})
+
+			t.LogStepf("Create namespace (%s,%s) and add it into SMMR", ns, anotherNs)
+			oc.CreateNamespace(t, ns, anotherNs)
+			updateDefaultSMMRWithNamespace(t, ns, anotherNs)
+
+			t.LogStepf("Update SMMR with only %s", ns)
+			updateDefaultSMMRWithNamespace(t, ns)
+
+			t.LogStepf("Look for %s in prometheus ConfigMap", ns)
+			testPrometheusConfigWithAsserts(t, assertConfigMapContainsNamespace(ns), assertConfigMapDoesNotContainNamespace(anotherNs))
+		})
+
+		t.NewSubTest("when the default SMMR with no member").Run(func(t test.TestHelper) {
```

Comment:
This seems to be flaky.

Reply:
Reproduced it in CRC. For each sub-test, I try to clean up the test by resetting to the default SMMR and I didn't wait for this SMMR change get picked up by the Prometheus reloader. 

I moved two subtest which require to check the reload trigger message to be the first and last tests, which works around this issue. I specified 10 second for the reloader to watch for changes but somehow this 10 second is longer in the CRC than it is on the Quicklab.